build.sbt

```diff
@@ -47,6 +49,7 @@ val commonTestDockerSettings: Seq[Setting[_]] = Seq(
 lazy val clio = project
   .in(file("."))
   .settings(commonSettings)
+  .settings(commonNoPublishSettings)
```

Comment:
Any reason to not put these settings in `commonSettings`? I assume we can override the settings for the projects that need deployment enabled. Either that, or only add the publish plugin to projects that will be publishing.

Reply:
If I put them in common settings, I think the order-of-operations for applying settings will cause them to override the publishing plugin's settings in the projects that need it (since plugins are applied before anything in build.sbt).

These settings aren't really necessary as long as you don't run `sbt publish` from the top-level project. If you do without these settings, SBT will try to publish everything, and complain about missing settings in the projects without the plugin. These settings tell SBT "no really, don't even try to publish this".